src/client/app/components/DashboardComponent.tsx

```diff
@@ -11,60 +11,132 @@ import MultiCompareChartContainer from '../containers/MultiCompareChartContainer
-	let showSpinner = false;
-	if (props.chartToRender === ChartTypes.line) {
-		if (props.lineLoading) {
-			showSpinner = true;
-		}
-		ChartToRender = LineChartContainer;
-	} else if (props.chartToRender === ChartTypes.bar) {
-		if (props.barLoading) {
-			showSpinner = true;
-		}
-		ChartToRender = BarChartContainer;
-	} else {
-		if (props.compareLoading) {
-			showSpinner = true;
-		}
-		ChartToRender = MultiCompareChartContainer;
+export default class DashboardComponent extends React.Component<DashboardProps, {}> {
+	constructor(props: DashboardProps) {
+		super(props);
+		this.handleTimeIntervalChange = this.handleTimeIntervalChange.bind(this);
 	}
 
-	const optionsClassName = props.optionsVisibility ? 'col-2 d-none d-lg-block' : 'd-none';
-	const chartClassName = props.optionsVisibility ? 'col-12 col-lg-10' : 'col-12';
+	public render(){
+		let ChartToRender: typeof LineChartContainer | typeof MultiCompareChartContainer | typeof BarChartContainer;
+		let showSpinner = false;
+		if (this.props.chartToRender === ChartTypes.line) {
+			if (this.props.lineLoading) {
+				showSpinner = true;
+			}
+			ChartToRender = LineChartContainer;
+		} else if (this.props.chartToRender === ChartTypes.bar) {
+			if (this.props.barLoading) {
+				showSpinner = true;
+			}
+			ChartToRender = BarChartContainer;
+		} else {
+			if (this.props.compareLoading) {
+				showSpinner = true;
+			}
+			ChartToRender = MultiCompareChartContainer;
+		}
 
-	return (
-		<div className='container-fluid'>
-			<div className='row'>
-				<div className={optionsClassName}>
-					<UIOptionsContainer />
-				</div>
-				<div className={`${chartClassName} align-self-center text-center`}>
-					{ showSpinner ? (
-						<SpinnerComponent loading width={50} height={50} />
-					) : (
-						<ChartToRender />
-					)}
+		const optionsClassName = this.props.optionsVisibility ? 'col-2 d-none d-lg-block' : 'd-none';
+		const chartClassName = this.props.optionsVisibility ? 'col-12 col-lg-10' : 'col-12';
+
+		const buttonMargin: React.CSSProperties = {
+			marginRight: '10px'
+		};
+
+		return (
+			<div className='container-fluid'>
+				<div className='row'>
+					<div className={optionsClassName}>
+						<UIOptionsContainer />
+					</div>
+					<div className={`${chartClassName} align-self-center text-center`}>
+						{ showSpinner ? (
+							<SpinnerComponent loading width={50} height={50} />
+						) : (
+							<ChartToRender />
+						)}
+						{ (this.props.chartToRender === ChartTypes.line) ? (
+							[<Button key={1}
+								style={buttonMargin}
+								onClick={() => this.handleTimeIntervalChange("range")}
+							> Redraw
+							</Button>,
+							<Button key={2}
+								style={buttonMargin}
+								onClick={() => this.handleTimeIntervalChange("all")}
+							> Restore
+							</Button>]
+						) : (
+							null
+						)}
+					</div>
 				</div>
 			</div>
-		</div>
-	);
+		);		
+	}
+
+	private handleTimeIntervalChange(mode: string) {
```

Comment:
I wonder if mode should be an enum since it has clearly defined values. Also, since you use if/else below you could use a boolean but that implies there will only be two possible values forever (I think this may be the case).